platforms/nuget/build-nuspec.py

```diff
@@ -0,0 +1,130 @@
+                                     usage='')
+    # Main arguments
+    parser.add_argument("--package_name", required=True, help="Package name. e.g.: OpenCV.CPP")
+    parser.add_argument("--package_version", required=True, help="Package version. e.g: 1.0.0")
+    parser.add_argument("--output_path", required=True, help="Nuget packages output directory - this is where target, nuspec, etc go.")
+    parser.add_argument("--sources_path", required=True, help="OpenCV source code root.")
+    parser.add_argument("--targets_path", required=True, help="Path name for the generated targets file.")
+    parser.add_argument("--targets_file", required=True, help="File name for the generated targets file.")
+
+    return parser.parse_args()
+
+
+
+def generate_files(list, args):
+    files_list = ['<files>']
+
+    # add source includes folder
+    derived_path = os.path.join(args.sources_path, 'include\**\*.*')
+    files_list.append(f'<file src="{os.path.join(derived_path)}" target="build\\native\\include" />')
+
+    # add vc14 compiled lib & dlls
+    derived_path = os.path.join(args.sources_path, 'x64\\vc14\\lib\\**\\opencv_world*.lib')
+    files_list.append(f'<file src="{os.path.join(derived_path)}" target="build\\native\\x64\\vc14.0\\lib" />')
+    derived_path = os.path.join(args.sources_path, 'x64\\vc14\\bin\\**\\opencv_world*.dll')
+    files_list.append(f'<file src="{os.path.join(derived_path)}" target="build\\native\\x64\\vc14.0\\bin" />')
+    derived_path = os.path.join(args.sources_path, 'x64\\vc14\\bin\\**\\opencv_world*.pdb')
+    files_list.append(f'<file src="{os.path.join(derived_path)}" target="build\\native\\x64\\vc14.0\\bin" />')
+
+    # add vc15 compiled lib & dlls
+    derived_path = os.path.join(args.sources_path, 'x64\\vc15\\lib\\**\\opencv_world*.lib')
+    files_list.append(f'<file src="{os.path.join(derived_path)}" target="build\\native\\x64\\vc15.0\\lib" />')
+    derived_path = os.path.join(args.sources_path, 'x64\\vc15\\bin\\**\\opencv_world*.dll')
+    files_list.append(f'<file src="{os.path.join(derived_path)}" target="build\\native\\x64\\vc15.0\\bin" />')
+    derived_path = os.path.join(args.sources_path, 'x64\\vc15\\bin\\**\\opencv_world*.pdb')
+    files_list.append(f'<file src="{os.path.join(derived_path)}" target="build\\native\\x64\\vc15.0\\bin" />')
+
+    # add vc16 compiled lib & dlls
+    derived_path = os.path.join(args.sources_path, 'x64\\vc16\\lib\\**\\opencv_world*.lib')
+    files_list.append(f'<file src="{os.path.join(derived_path)}" target="build\\native\\x64\\vc16.0\\lib" />')
+    derived_path = os.path.join(args.sources_path, 'x64\\vc16\\bin\\**\\opencv_world*.dll')
+    files_list.append(f'<file src="{os.path.join(derived_path)}" target="build\\native\\x64\\vc16.0\\bin" />')
+    derived_path = os.path.join(args.sources_path, 'x64\\vc16\\bin\\**\\opencv_world*.pdb')
+    files_list.append(f'<file src="{os.path.join(derived_path)}" target="build\\native\\x64\\vc16.0\\bin" />')
+
```

Comment:
It makes sense to do it in a loop by list of VS versions: ['vc14', 'vc15', 'vc16'] and substitute VS version to path.

Reply:
This change is done in the last commit.